src/main/java/apoc/export/cypher/formatter/CypherFormatterUtils.java

```diff
@@ -0,0 +1,185 @@
+package apoc.export.cypher.formatter;
+
+import apoc.export.util.FormatUtils;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.PropertyContainer;
+import org.neo4j.helpers.collection.Iterables;
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+/**
+ * @author AgileLARUS
+ *
+ * @since 16-06-2017
+ */
+public class CypherFormatterUtils {
+
+	public final static String UNIQUE_ID_LABEL = "UNIQUE IMPORT LABEL";
+	public final static String UNIQUE_ID_PROP = "UNIQUE IMPORT ID";
+	public final static String Q_UNIQUE_ID_LABEL = quote(UNIQUE_ID_LABEL);
+
+    // ---- common ----
+
+    public static String quote(String id) {
+        return "`" + id + "`";
+    }
+
+    public static String label(String id) {
+        return ":`" + id + "`";
+    }
+
+    // ---- label ----
+
+	public static String labelsAsString(Node node, Map<String, String> uniqueConstraints, Set<String> indexNames) {
+		Iterator<Label> labels = node.getLabels().iterator();
+		StringBuilder result = new StringBuilder(100);
+		boolean uniqueFound = false;
+		while (labels.hasNext()) {
+			Label next = labels.next();
+			String labelName = next.name();
+			if (uniqueConstraints.containsKey(labelName) && node.hasProperty(uniqueConstraints.get(labelName)))
+				uniqueFound = true;
+			if (indexNames.contains(labelName))
+				result.insert(0, label(labelName));
+			else
+				result.append(label(labelName));
+		}
+		if (!uniqueFound) {
+			result.append(label(UNIQUE_ID_LABEL));
+		}
+		return result.toString();
+	}
+
+    public static String notUniqueLabelsAsString(String id, Node node, Map<String, String> uniqueConstraints) {
+
+        String nodeLookupLabel = null;
+        for (Label l : node.getLabels()) {
+            String label = l.name();
+            String prop = uniqueConstraints.get(label);
+            if (prop == null) continue;
+            Object value = node.getProperty(prop, null);
+            if (value == null) continue;
+            nodeLookupLabel = label;
+        }
+
+        Iterator<Label> labels = node.getLabels().iterator();
+        StringBuilder result = new StringBuilder(100);
+        while (labels.hasNext()) {
+            Label next = labels.next();
+            String labelName = next.name();
+            if (labelName.equals(nodeLookupLabel)) continue;
+            result.append(", " + id + label(labelName));
```

Comment:
where is the output of this used? it seems weird to have `id:Label1, id:Label2` as output?